lib/workflow/stores/mongo.js

```diff
@@ -0,0 +1,563 @@
+// Copyright 2015, EMC, Inc.
+'use strict';
+
+module.exports = mongoStoreFactory;
+mongoStoreFactory.$provide = 'TaskGraph.Stores.Mongo';
+mongoStoreFactory.$inject = [
+    'Services.Waterline',
+    'Promise',
+    'Constants',
+    'Errors',
+    'Assert',
+    '_'
+];
+function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
+    var exports = {};
+
+    // NOTE: This is meant to be idempotent, and just drop the update silently
+    // if the graph has already been marked as done elsewhere and the query returns
+    // empty.
+    exports.setGraphDone = function(state, data) {
```

Comment:
since this appears to be the primary (and only) interface point into the backend stores for the workflow atomicity, could you add JSDoc to these `exports` methods that describe the inputs and return values expected